fsw/standalone/generic_reaction_wheel_checkout.c

```diff
@@ -0,0 +1,249 @@
+
+/*
+** Include Files
+*/
+#include "generic_reaction_wheel_checkout.h"
+
+/*
+** Global Variables
+*/
+static uart_info_t RW_UART[3] = {
+    {.deviceString = &GENERIC_REACTION_WHEEL_1_CFG_STRING[0], .handle = GENERIC_REACTION_WHEEL_1_CFG_HANDLE, .isOpen = GENERIC_REACTION_WHEEL_1_CFG_IS_OPEN, .baud = GENERIC_REACTION_WHEEL_1_CFG_BAUDRATE_HZ},
+    {.deviceString = &GENERIC_REACTION_WHEEL_2_CFG_STRING[0], .handle = GENERIC_REACTION_WHEEL_2_CFG_HANDLE, .isOpen = GENERIC_REACTION_WHEEL_2_CFG_IS_OPEN, .baud = GENERIC_REACTION_WHEEL_2_CFG_BAUDRATE_HZ},
+    {.deviceString = &GENERIC_REACTION_WHEEL_3_CFG_STRING[0], .handle = GENERIC_REACTION_WHEEL_3_CFG_HANDLE, .isOpen = GENERIC_REACTION_WHEEL_3_CFG_IS_OPEN, .baud = GENERIC_REACTION_WHEEL_3_CFG_BAUDRATE_HZ},
+};
+int32_t RwData;
+
+/*
+** Component Functions
+*/
+void print_help(void) 
+{
+    printf(PROMPT "command [args]\n"
+        "---------------------------------------------------------------------\n"
+        "help                               - Display help                    \n"
+        "exit                               - Exit app                        \n"
+        "get_momentum                      - Request rw momentum            \n"
+        "  m                                - ^                               \n"
+        "set_torque x y z                   - Send torque for RWs             \n"
+        "  t x y z                          - ^                               \n"
+        "\n"
+    );
+}
+
+
+int get_command(const char* str)
+{
+    int status = CMD_UNKNOWN;
+    char lcmd[MAX_INPUT_TOKEN_SIZE];
+    strncpy(lcmd, str, MAX_INPUT_TOKEN_SIZE);
+
+    /* Convert command to lower case */
+    to_lower(lcmd);
+
+    if(strcmp(lcmd, "help") == 0) 
+    {
+        status = CMD_HELP;
+    }
+    else if(strcmp(lcmd, "exit") == 0) 
+    {
+        status = CMD_EXIT;
+    }
+    else if(strcmp(lcmd, "get_momentum") == 0) 
+    {
+        status = CMD_GET_MOMENTUM;
+    }
+    else if(strcmp(lcmd, "m") == 0) 
+    {
+        status = CMD_GET_MOMENTUM;
+    }
+    else if(strcmp(lcmd, "set_torque") == 0) 
+    {
+        status = CMD_SET_TORQUE;
+    }
+    else if(strcmp(lcmd, "t") == 0) 
+    {
+        status = CMD_SET_TORQUE;
+    }
+    return status;
+}
+
+
+int process_command(int cc, int num_tokens, char tokens[MAX_INPUT_TOKENS][MAX_INPUT_TOKEN_SIZE])
+{
+    int32_t status = OS_SUCCESS;
+    int32_t exit_status = OS_SUCCESS;
+    uint32_t torque;
+
+    /* Process command */
+    switch(cc) 
+    {	
+        case CMD_HELP:
+            print_help();
+            break;
+        
+        case CMD_EXIT:
+            exit_status = OS_ERROR;
+            break;
+
+        case CMD_GET_MOMENTUM:
+            if (check_number_arguments(num_tokens, 0) == OS_SUCCESS)
+            {
+                for ( int i = 0; i < 3; i++)
+                {
+                    status = GetCurrentMomentum(&RW_UART[i], &RwData);
```

Comment:
![image](https://github.com/user-attachments/assets/59ce6ee3-6bfc-495a-8d30-96addd2512bb)

Reply:
Fixed that. Also fixed it using int32s instead of doubles. I had switched it, because YAMCS seems to treat it as int32s, but I believe it was doubles originally (can change back if wrong)

Reply:
@jlucas9